.evergreen/scripts/find-cmake-version.sh

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env bash
```

Comment:
This has a lot of functionality overlap with the CMake scripts that are used in `libmongocrypt`. They should probably be consolidated.

Reply:
Given libmongocrypt unconditionally uses a copy of the C Driver but not the reverse (libmongocrypt only required when CSFLE is enabled + building from source), I think consolidation would involve libmongocrypt utilizing (in some part or all) CMake download scripts in the C Driver rather than the reverse.

Personally, I am of the mind that given libmongocrypt and libmongoc are sufficiently distinct in their compile and test coverage requirements, I do not think each repo having its own download scripts (even if seemingly redundant/overlapping) is an issue. I would prefer limiting (circular) interdependencies if able.

Reply:
I'm not necessarily suggesting that one should depend on the other (although that may be useful if CI complexity were to grow), but that these operations can be subtle and it may be more clear to use the same code for both, even if that's done by copy-pasting some scripts between them.

Reply:
Consolidating libmongocrypt's `ensure-cmake.sh` and the C Driver `find-cmake-version.sh` seems preferable, though I do not consider it a blocker to merge.

Reply:
It's something to possibly consider at a later time, as its not really too concerning (for now).